pinot-common/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -102,3 +102,20 @@ SqlNodeList SqlStmtsEof() :
         return stmts;
     }
 }
+
+void SqlAtTimeZone(List<Object> list, ExprContext exprContext, Span s) :
+{
+    List<Object> list2;
+    SqlOperator op;
+}
+{
+    {
+        checkNonQueryExpression(exprContext);
+        s.clear().add(this);
+    }
```

Comment:
why do we need to check this. can't it be something like
```
SELECT * FROM tbl 
WHERE (SELECT MAX(timestampWithoutTimeZone) FROM anotherTbl) AT TIME ZONE 'Pacific' BETWEEN xxx AND xxx
```

Reply:
I modeled this after the LIKE syntax which has that check - I'm not 100% sure why they added that restriction, because I checked Postgres and `SELECT (SELECT str FROM foo) LIKE 'foo' FROM foo;` works and so does nested `AT TIME ZONE` like your example query.

I tried these in Pinot, though, and it looks like there are other limitations that prevent us from doing that...

`SELECT (SELECT str FROM foo) LIKE 'foo' FROM foo;` returns the following exception in Pinot:
```
java.lang.ClassCastException: class org.apache.calcite.sql.SqlSelect cannot be cast to class org.apache.calcite.sql.SqlBasicCall (org.apache.calcite.sql.SqlSelect and org.apache.calcite.sql.SqlBasicCall are in unnamed module of loader 'app')

	at org.apache.pinot.sql.parsers.CalciteSqlParser.toExpression(CalciteSqlParser.java:748)
	at org.apache.pinot.sql.parsers.CalciteSqlParser.compileFunctionExpression(CalciteSqlParser.java:806)
	at org.apache.pinot.sql.parsers.CalciteSqlParser.toExpression(CalciteSqlParser.java:748)
```

If you feel strongly, I can remove the check, but it won't work either way without additional work (which I think we can address in a follow-up if desired)